server/src/com/cloud/api/ApiResponseHelper.java

```diff
@@ -479,6 +479,17 @@ public SnapshotResponse createSnapshotResponse(Snapshot snapshot) {
             if (zone != null) {
                 snapshotResponse.setZoneId(zone.getUuid());
             }
+
+            if (volume.getVolumeType().equals(Volume.Type.ROOT)) {
+                VMInstanceVO instance = ApiDBUtils.findVMInstanceById(volume.getInstanceId());
+                if (instance != null) {
+                    GuestOS guestOs = ApiDBUtils.findGuestOSById(instance.getGuestOSId());
```

Comment:
It feels like lines 484 & 486 could be combined into a single join exposed by a method on ``VolumeDao``.  Do you agree?